scripts/fastsp_script/fsp_test.py

```diff
@@ -1,114 +1,110 @@
-                cmd = "java -jar FastSP.jar -r " + fasta1 + " -e " + fasta2 + " > " + infile_name
-                cmd_exit_code = os.system(cmd)
-                assert(cmd_exit_code == 0), "java fastsp failed to run"
-          
-            
-                with open(infile_name, "r") as infile:
-                    ##HERE COMES THE STUFF
-                    ## IT'S AUTOCLOSED OUTSIDE THE WITH BLOCK
-                    if_lines = infile.readlines()
-                ## call a function here after file is closed
-                parse_fastsp(if_lines, dataset, species, datatype, ref_num, est_num):
-
-
-                ## IN ONLY THE NEXT TWO LINES! we open in APPEND mode, write the line, and close the file automatically
-                with open(outfile_name, "a") as outfile:
-                    outfile.write(write_line)
+    aln_dir = [x for x in os.listdir(overall_path) if os.path.isdir(os.path.join(overall_path,x))]
+    
+    counter = 0
+    for dir in aln_dir:
+        file_path = os.path.join(overall_path,aln_dir[counter])
+        counter +=1
+        fas_files = [x for x in os.listdir(file_path) if x.endswith(FAS)]  
+           
+        for i in range(n_alignments):
+            file1  = fas_files[i]
+            fasta1 = file_path+SLASH+file1
+            file1_split = file1.split("_")
+            file1_dot = file1.split(".")
+            dataset = file1_dot[0]
+            #print(dataset)
+            species = file1_dot[1]
+            datatype = file1_split[1]
+            dum_ref_num = file1_split[2]
+            ref_num = dum_ref_num.split(".")[0]
+            #print(ref_num)    
+           
+            for j in range (i,n_alignments):
+                if i != j:
+                    file2 = fas_files[j] 
+                    fasta2 = file_path+SLASH+file2
+                    file2_split = file2.split("_")
+                    dum_est_num = file2_split[2]
+                    est_num = dum_est_num.split(".")[0]
+                    #print(est_num)
                     
+                    version_infile = tempfile.NamedTemporaryFile()
                     
+    
+                    ## REMINDER: COMMENT OUT PRINT LINE WHEN TOTALLY FINISHED
+                    #print("Running FastSP analysis on " + file1 + " and " + file2 + "\n")
+                    cmd = "java -jar FastSP.jar -r " + fasta1 + " -e " + fasta2 + " > " + version_infile.name
+                    cmd_exit_code = os.system(cmd)
+                    assert(cmd_exit_code == 0), "java fastsp failed to run"
+              
+                
+                    with open(version_infile.name, "r") as infile:
+                        ##HERE COMES THE STUFF
+                        ## IT'S AUTOCLOSED OUTSIDE THE WITH BLOCK
+                        if_lines = infile.readlines()
+                    ## call a function here after file is closed
+                    parse_fastsp(if_lines, dataset, species, datatype, ref_num, est_num)
```

Comment:
`write_line = parse_fastsp(.....)`. You returned the string but you never SAVED it to a variable!! This is why you did the `global` thing. Don't do the global thing ever. Just define the returned value.